### PR TITLE
Develop file parse

### PR DESCRIPTION
**Pull request by mfdeveloper508**
- add file interface
- add file parse
- add marshal/unmarshal of json
- add test files
- add string

### CODE REVIEW

File: file/file_instance.go [hunk 1]
```diff
@@ -0,0 +1,325 @@
...
+
+// MarshalJSON returns JSON blob
+func (f *fileInstance) MarshalJSON() ([]byte, error) {
+	buffer := bytes.NewBufferString("{")
+
+	value, err := json.Marshal(f.Header)
+	if err != nil {
+		return nil, err
+	}
+	buffer.WriteString(fmt.Sprintf("\"%s\":%s,", f.Header.Description(), string(value)))
```
Comment (adamdecaf) [inline file/file_instance.go:152]: Will the struct tags not marshal properly here? We can ignore fields with `json:"-"` or override more MarhsalJSON funcs in structs. 

I don't like the `fmt.Sprintf(..)` calls here.

File: utils/file.go [hunk 2]
```diff
@@ -0,0 +1,18 @@
+package utils
+
+import (
+	"bufio"
+	"os"
+	"strings"
+)
+
+func ReadStringFromFile(f *os.File) string {
```
Comment (adamdecaf) [inline utils/file.go:9]: What if we called this `ReadFile` in the root `metro2` project?

File: file/file_instance.go (hunk 1 above)
Comment (mfdeveloper508) [reply file/file_instance.go]: updated the logic

File: utils/file.go (hunk 2 above)
Comment (mfdeveloper508) [reply utils/file.go]: changed

**Pull request merged**